sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs

```diff
@@ -137,14 +138,18 @@ public static partial class TablesModelFactory
     public partial class TableTransactionalBatch
     {
         protected TableTransactionalBatch() { }
+        public TableTransactionalBatch(string partitionKey) { }
+        public int OperationCount { get { throw null; } }
         public virtual void AddEntities<T>(System.Collections.Generic.IEnumerable<T> entities) where T : class, Azure.Data.Tables.ITableEntity, new() { }
-        public virtual void AddEntity<T>(T entity) where T : class, Azure.Data.Tables.ITableEntity, new() { }
+        public virtual void AddEntity(Azure.Data.Tables.ITableEntity entity) { }
+        public void ClearOperations() { }
         public virtual void DeleteEntity(string rowKey, Azure.ETag ifMatch = default(Azure.ETag)) { }
-        public virtual Azure.Response<Azure.Data.Tables.Models.TableBatchResponse> SubmitBatch(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.Data.Tables.Models.TableBatchResponse>> SubmitBatchAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public System.Collections.Generic.IReadOnlyList<Azure.Data.Tables.ITableEntity> GetEntities() { throw null; }
+        public bool RemoveEntityOperation(Azure.Data.Tables.ITableEntity entity) { throw null; }
+        public bool RemoveEntityOperations(System.Collections.Generic.IEnumerable<Azure.Data.Tables.ITableEntity> entities) { throw null; }
         public virtual bool TryGetFailedEntityFromException(Azure.RequestFailedException exception, out Azure.Data.Tables.ITableEntity failedEntity) { throw null; }
-        public virtual void UpdateEntity<T>(T entity, Azure.ETag ifMatch, Azure.Data.Tables.TableUpdateMode mode = Azure.Data.Tables.TableUpdateMode.Merge) where T : class, Azure.Data.Tables.ITableEntity, new() { }
-        public virtual void UpsertEntity<T>(T entity, Azure.Data.Tables.TableUpdateMode mode = Azure.Data.Tables.TableUpdateMode.Merge) where T : class, Azure.Data.Tables.ITableEntity, new() { }
+        public virtual void UpdateEntity(Azure.Data.Tables.ITableEntity entity, Azure.ETag ifMatch, Azure.Data.Tables.TableUpdateMode mode = Azure.Data.Tables.TableUpdateMode.Merge) { }
+        public virtual void UpsertEntity(Azure.Data.Tables.ITableEntity entity, Azure.Data.Tables.TableUpdateMode mode = Azure.Data.Tables.TableUpdateMode.Merge) { }
```

Comment:
How do we choose when to provide only singular overload and when the IEnumerable one?

Reply:
It's mostly an assumption that add is the common bulk scenario. We can always add more later if needed.